### DISCUSSION TITLE
What am I doing wrong? The gap between two drivers during qualifying

### DISCUSSION THREAD

**Discussion by MooshuBeef** [theOehrly/Fast-F1#350] [Q&A] [answered]
I adapted the code from @laurence9899 [from this post](https://github.com/theOehrly/Fast-F1/discussions/347) and want to verify the visualization of the gap between two drivers is correct. 

The fastest lap times during qualifiers between Russell and Hamilton in the AUS GP 2023 were correct as follows when calling it via the API: 
`HAM Fastest lap from Qualifiers: 0 days 00:01:17.104000`
`RUS Fastest lap from Qualifiers: 0 days 00:01:16.968000`

That should be a gap of ~0.1s. But then my plot shows it at ~0.3s. What am I doing wrong? 

![image](https://user-images.githubusercontent.com/27934886/230498897-90f3bd59-7fcb-4683-81d8-0ec7153bdf95.png)

```import fastf1 as ff1
from fastf1 import plotting
from fastf1 import utils

from matplotlib import pyplot as plt
from matplotlib.pyplot import figure

import numpy as np
import pandas as pd
ff1.Cache.enable_cache('~/Documents/Formula1_python/venv/cache') # replace with your cache directory

# Setup plotting

plotting.setup_mpl()

year, grand_prix, session = 2023, 'AUSTRALIA', 'Q'

quali = ff1.get_session(year, grand_prix, session)
quali.load() # This is new with Fastf1 v.2.2

# This is how it used to be:

# laps = quali.load(with_telemetry=True)

driver_1, driver_2 = 'HAM', 'RUS'

# Laps can now be accessed through the .laps object coming from the session

laps_driver_1 = quali.laps.pick_driver(driver_1)
laps_driver_2 = quali.laps.pick_driver(driver_2)

# Select the fastest lap

fastest_driver_1 = laps_driver_1.pick_fastest()
fastest_driver_2 = laps_driver_2.pick_fastest()

# Retrieve the telemetry and add the distance column

telemetry_driver_1 = fastest_driver_1.get_telemetry().add_distance()
telemetry_driver_2 = fastest_driver_2.get_telemetry().add_distance()

# Make sure whe know the team name for coloring

team_driver_1 = fastest_driver_1['Team']
team_driver_2 = fastest_driver_2['Team']

#Set Colors

driver_1_color = 'cyan'
driver_2_color = 'orange'

# Extract the delta time

delta_time, ref_tel, compare_tel = utils.delta_time(fastest_driver_1, fastest_driver_2)

plot_size = [15, 8]
plot_title = f"{quali.event.year} {quali.event.EventName} - {quali.name} - {driver_1} VS {driver_2}"
plot_filename = plot_title.replace(" ", "") + ".png"

# Make plot a bit bigger

plt.rcParams['figure.figsize'] = plot_size

# Set the plot title

plt.title(plot_title)

# Delta line

plt.plot(ref_tel['Distance'], delta_time, label=driver_1, color=driver_1_color)
plt.axhline(0, label=driver_2, color=driver_2_color)
plt.set(ylabel=f"Gap to {driver_2} (s)")
plt.legend(loc="lower left")
plt.grid()

# Store figure

plt.savefig(plot_filename, dpi=300)
plt.show()
```

**Comment by theOehrly** [comment, marked as answer]
Quoting from the docs for `utils.delta_time`:

> This is a nice gimmick but not actually very accurate which is an inherent problem from the way this is calculated currently (There may not be a better way though). In comparison with the sector times and the differences that can be calculated from these, there are notable differences! You should always verify the result against sector time differences or find a different way for verification.

Looks like you are not actually doing anything wrong here, judging by a quick check. It's just that the calculation result seems to be way off for this combination of laps. 
The delta time feature might either get removed or completely rewritten in the future. [f1-tempo.com](https://www.f1-tempo.com) has its own delta time implementation that seems to work much better. So there seems to be a way to make it work at least.
